core/encoding/json/parser.odin

```diff
@@ -263,7 +264,12 @@ parse_object_body :: proc(p: ^Parser, end_token: Token_Kind) -> (obj: Object, er
 			return
 		}
 
-		obj[key] = elem
+		insert_success := runtime.map_insert(&obj, key, elem)
```

Comment:
@Kelimion Do you think it would be advisable here to use `map_grow_...` or `reserve` instead? Is there a good way to get the allocation error more specifically?

There is a test failing in the build because of this added code but it's fairly hard to see why 190 extra entries would be added to the parsed array in the test.